### PR TITLE
fix: use document over window for browser check

### PR DESCRIPTION
**Pull request by jacob-ebey**
Runtimes such as Deno have a window defined making this check not work.

### CODE REVIEW

File: docs/other-api/constraints.md [hunk 1]
```diff
@@ -248,13 +248,13 @@ The most common scenario is intitializing a third party API when your module is
 
 #### Window Guard
```
Comment (mjackson) [inline docs/other-api/constraints.md:249]: Let's change this to say "Document Guard"

File: docs/other-api/constraints.md [hunk 2]
```diff
@@ -248,13 +248,13 @@ The most common scenario is intitializing a third party API when your module is
...
 
-This ensures the library is only initialized if there is a `window`, meaning you're in the browser.
+This ensures the library is only initialized if there is a `document`, meaning you're in the browser. We recomend `document` over `window` because server runtimes like Deno have a global `window` available.
```
Comment (mjackson) [inline docs/other-api/constraints.md:251]: s/server runtimes like Deno have/Deno has/

**Review by mjackson — CHANGES_REQUESTED** [review]

**Review by mcansh — APPROVED** [review]

**Review by kentcdodds — APPROVED** [review]
Perfect

**Comment by btargac** [comment]
Does remix run on Deno instead of Node runtime ?

**Comment by girishk21** [comment]
> Does remix run on Deno instead of Node runtime ?

https://twitter.com/ebey_jacob/status/1471009211416932353?s=20 it does, and it runs really well 😄 . The official adapter should land soon I guess

**Comment by kentcdodds** [comment]
It runs on both and more so we need code that runs equally well on all platforms we support.

**Pull request merged**